workload/terraform/modules/mgmtvm/main.tf

```diff
@@ -0,0 +1,144 @@
+# Resource block to generate a random string for the local admin password
+resource "random_string" "AVD_local_password" {
+  length           = 16
+  special          = true
+  min_special      = 2
+  override_special = "*!@#?"
+}
+
+# User Assigned Managed Identity for the MGMT VM to join storage accounts to the domain
+resource "azurerm_user_assigned_identity" "stguai" {
+  name                = "id-storage-${var.prefix}-${var.deployment_environment}-001"
+  location            = data.azurerm_resource_group.rg.location
+  resource_group_name = data.azurerm_resource_group.rg.name
+}
+
+# Assigns the Managed Identity Storage Account Contributor RBAC to the Storage RG scope
+resource "azurerm_role_assignment" "assign_stguai_sac" {
+  scope                = data.azurerm_resource_group.rg_storage.id
+  role_definition_name = "Storage Account Contributor"
+  principal_id         = azurerm_user_assigned_identity.stguai.principal_id
+}
+
+# Assigns the Managed Identity Reader RBAC to the Storage RG scope
+resource "azurerm_role_assignment" "assign_stguai_sac" {
```

Comment:
Duplicated resource name "assign_stguai_sac"

Reply:
addressed - changed `sac` in code to `sar`